RoddyCore/test/de/dkfz/roddy/tools/ComplexLineTest.groovy

```diff
@@ -0,0 +1,123 @@
+        assert ComplexLine.isOpeningOrClosingCharacter('}' as Character);
+        assert ComplexLine.isOpeningOrClosingCharacter('(' as Character);
+        assert ComplexLine.isOpeningOrClosingCharacter(')' as Character);
+    }
+
+    @Test
+    public void testFitsToOpeningCharacter() {
+        assert ComplexLine.fitsToOpeningCharacter('\'' as Character, '\'' as Character)
+        assert ComplexLine.fitsToOpeningCharacter('\"' as Character, '\"' as Character)
+        assert !ComplexLine.fitsToOpeningCharacter('\"' as Character, '\'' as Character)
+        assert ComplexLine.fitsToOpeningCharacter('}' as Character, '{' as Character)
+        assert ComplexLine.fitsToOpeningCharacter(']' as Character, '[' as Character)
+        assert ComplexLine.fitsToOpeningCharacter(')' as Character, '(' as Character)
+    }
+
+    @Test
+    public void testParseSimpleLine() {
+        ComplexLine.ComplexNode parsed = ComplexLine.parseLine("a'bcd'e f'gh'i") as ComplexLine.ComplexNode
+        assert parsed.blocks.size() == 5
+        assert parsed.blocks[0] instanceof ComplexLine.TextNode && parsed.blocks[0].reassemble() == "a'"
+        assert parsed.blocks[1] instanceof ComplexLine.ComplexNode && parsed.blocks[1].reassemble() == "bcd"
+        assert parsed.blocks[2] instanceof ComplexLine.TextNode && parsed.blocks[2].reassemble() == "'e f'"
+        assert parsed.blocks[3] instanceof ComplexLine.ComplexNode && parsed.blocks[3].reassemble() == "gh"
+        assert parsed.blocks[4] instanceof ComplexLine.TextNode && parsed.blocks[4].reassemble() == "'i"
+    }
+
+    @Test
+    public void testParseCommandLine() {
+        ComplexLine.ComplexNode parsed = ComplexLine.parseLine(commandLineExample) as ComplexLine.ComplexNode;
+        assert parsed.blocks.size() == 5
+        assert parsed.blocks[0] instanceof ComplexLine.TextNode && parsed.blocks[0].reassemble() == "--abc --def=\""
+        assert ((ComplexLine.ComplexNode)parsed.blocks[1]).blocks[0].reassemble() == "ab,def;efk=test;anotherValue='"
+        assert parsed.blocks[2] instanceof ComplexLine.TextNode && parsed.blocks[2].reassemble() == "\" --jki --a=\""
+        assert parsed.blocks[3] instanceof ComplexLine.ComplexNode && parsed.blocks[3].reassemble() == "something=def"
+        assert parsed.blocks[4] instanceof ComplexLine.TextNode && parsed.blocks[4].reassemble() == "\""
+    }
+
+    @Test
+    public void testParseCodeLine() {
+        ComplexLine.ComplexNode parsed = ComplexLine.parseLine(codeLineExample) as ComplexLine.ComplexNode;
+        assert parsed.blocks.size() == 7
+        assert parsed.blocks[0] instanceof ComplexLine.TextNode && parsed.blocks[0].reassemble() == "def multiline='"
+        assert ((ComplexLine.ComplexNode)parsed.blocks[1]).blocks[0].reassemble() == "a string"
+        assert parsed.blocks[2] instanceof ComplexLine.TextNode && parsed.blocks[2].reassemble() == "', b={"
+    }
+
+    @Test(expected = IOException)
```

Comment:
Can you make tests that also check the exception message? Look an @Rule.
Also: More faulty examples (see above)!